MediaBrowser.Providers/Lyric/LrcLyricProvider.cs

```diff
@@ -0,0 +1,211 @@
+            var timeData = sortedLyricData[i].TimeTags.First().Value;
+            if (timeData is null)
+            {
+                continue;
+            }
+
+            long ticks = TimeSpan.FromMilliseconds(timeData.Value).Ticks;
+            lyricList.Add(new LyricLine(sortedLyricData[i].Text, ticks));
+        }
+
+        if (fileMetaData.Count != 0)
+        {
+            // Map metaData values from LRC file to LyricMetadata properties
+            LyricMetadata lyricMetadata = MapMetadataValues(fileMetaData);
+
+            return new LyricResponse { Metadata = lyricMetadata, Lyrics = lyricList };
+        }
+
+        return new LyricResponse { Lyrics = lyricList };
+    }
+
+    /// <summary>
+    /// Converts metadata from an LRC file to LyricMetadata properties.
+    /// </summary>
+    /// <param name="metaData">The metadata from the LRC file.</param>
+    /// <returns>A lyricMetadata object with mapped property data.</returns>
+    private static LyricMetadata MapMetadataValues(IDictionary<string, string> metaData)
+    {
+        LyricMetadata lyricMetadata = new();
+
+        if (metaData.TryGetValue("ar", out var artist) && !string.IsNullOrEmpty(artist))
+        {
+            lyricMetadata.Artist = artist;
+        }
+
+        if (metaData.TryGetValue("al", out var album) && !string.IsNullOrEmpty(album))
+        {
+            lyricMetadata.Album = album;
+        }
+
+        if (metaData.TryGetValue("ti", out var title) && !string.IsNullOrEmpty(title))
+        {
+            lyricMetadata.Title = title;
+        }
+
+        if (metaData.TryGetValue("au", out var author) && !string.IsNullOrEmpty(author))
+        {
+            lyricMetadata.Author = author;
+        }
+
+        if (metaData.TryGetValue("length", out var length) && !string.IsNullOrEmpty(length))
+        {
+            if (DateTime.TryParseExact(length, _acceptedTimeFormats, null, DateTimeStyles.None, out var value))
+            {
+                lyricMetadata.Length = value.TimeOfDay.Ticks;
+            }
+        }
+
+        if (metaData.TryGetValue("by", out var by) && !string.IsNullOrEmpty(by))
+        {
+            lyricMetadata.By = by;
+        }
+
+        if (metaData.TryGetValue("offset", out var offset) && !string.IsNullOrEmpty(offset))
+        {
+            if (int.TryParse(offset, out var value))
+            {
+                lyricMetadata.Offset = TimeSpan.FromMilliseconds(value).Ticks;
+            }
+        }
+
+        if (metaData.TryGetValue("re", out var creator) && !string.IsNullOrEmpty(creator))
+        {
+            lyricMetadata.Creator = creator;
+        }
+
+        if (metaData.TryGetValue("ve", out var version) && !string.IsNullOrEmpty(version))
+        {
+            lyricMetadata.Version = version;
+        }
+
+        return lyricMetadata;
+    }
+
+    private static string GetMetadataFieldName(string metaDataRow, int index)
+    {
+        return metaDataRow.AsSpan(1, index - 1).Trim().ToString();
```

Comment:
I'm not sure but I think it might be better to check `IsEmpty` and then return `string.Empty` instead of calling `ToString` if the span is empty